File: youtubeextractor/src/test/java/com/commit451/youtubeextractor/ExtractionTest.java
package com.commit451.youtubeextractor;

import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

/**
 * To work on unit tests, switch the Test Artifact in the Build Variants view.
 */
@RunWith(RobolectricTestRunner.class)
@Config(constants = BuildConfig.class, sdk = 25)
public class ExtractionTest {

    private static final String GRID_YOUTUBE_ID = "9d8wWcJLnFI";

    @Test
    public void testExtraction() throws Exception {
        YouTubeExtractor extractor = YouTubeExtractor.create();
        YouTubeExtractionResult result = extractor.extract(GRID_YOUTUBE_ID)
                .blockingGet();
        Assert.assertNotNull(result);
    }
}

File: youtubeextractor/src/test/java/com/commit451/youtubeextractor/ExtractionTest.kt
package com.commit451.youtubeextractor

import org.junit.Assert
import org.junit.Test

class ExtractionTest {

    companion object {
        private const val YOUTUBE_ID_REQUIRES_SIGNATURE = "_r6CgaFNAGg"
    }

    @Test
    fun testExtractionWithSignature() {
        val extractor = YouTubeExtractor.Builder()
                .debug(true)
                .build()
        val startTime = System.currentTimeMillis()
        val result = extractor.extract(YOUTUBE_ID_REQUIRES_SIGNATURE)
                .blockingGet()
        println("Time taken: ${System.currentTimeMillis() - startTime}")
        testResult(result)
    }

    private fun testResult(extraction: YouTubeExtraction) {
        Assert.assertTrue(extraction.streams.isNotEmpty())
        Assert.assertTrue(extraction.thumbnails.isNotEmpty())
        Assert.assertNotNull(extraction.title)
        Assert.assertNotNull(extraction.description)
        Assert.assertNotNull(extraction.durationMilliseconds)
        Assert.assertNotNull(extraction.viewCount)
        Assert.assertNotNull(extraction.author)
    }
}
